src/opal.c

```diff
@@ -1508,12 +1508,33 @@ make_parentheses_expression(void)
 node_s*
 make_leaf(lexeme_type_e type, lexeme_s* curr_lexeme)
 {
-  /// Create the leaf node to return
-  node_s* node = NULL;
+    /// Create the leaf node to return
```

Comment:
In the current logic, the code will try to create a node with the int_val when the function is called with anything other than `lx_String` or `lx_Indent`. This may result in leaf nodes for error conditions. Can you also check for `lx_Integer` in an else if. That way, any other node type will return a `NULL` pointer and caller code can handle it appropriately.

Reply:
Thanks for the suggestions. All changes have been made and the PR re-pushed.